app/domain/person_tags/validation.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module PersonTags
+  class Validation
+
+    EMAIL_PRIMARY_INVALID='category_validation:email_primary_invalid'.freeze
+    EMAIL_ADDITIONAL_INVALID='category_validation:email_additional_invalid'.freeze
+
+    class << self
+
+      def tag_names
+        [EMAIL_PRIMARY_INVALID, EMAIL_ADDITIONAL_INVALID]
+      end
+
+      def list
+        [email_primary_invalid, email_additional_invalid]
+      end
+
+      def email_primary_invalid(create: false)
+        find(EMAIL_PRIMARY_INVALID, create: create)
+      end
+
+      def email_additional_invalid(create: false)
+        find(EMAIL_ADDITIONAL_INVALID, create: create)
+      end
+
+      private
+
+      def find(tag, create: false)
```

Comment:
The `create` switch completely determines the behaviour of this method, and in the code that uses it, it is always clear which one to use. Consider splitting it into two separate methods `find` and `create_if_not_exists`, both without the `create` argument.